### DISCUSSION TITLE
Implementing the Boiler Fire Side Model

### DISCUSSION THREAD

**Discussion by REI-Gardner** [IDAES/idaes-pse#536] [Q&A] [answered]
I am a new user to IDAES and I am interested in learning how to setup a simple flowsheet with the "Boiler Fire Side Model" implemented.

I have looked over the introduction information under the technical specifications [link](https://idaes-pse.readthedocs.io/en/stable/technical_specs/model_libraries/power_generation/unit_models/boiler_fireside.html). Are there additional resources for learning more about how this model works and how one would set it up? Specifically, the required user provided surrogate dictionary. Also is there any technical description of this model other than the above link?

**Comment by ksbeattie** [comment]
You might find more info by looking at the following notebook: https://github.com/IDAES/idaes-pse/blob/main/idaes/power_generation/flowsheets/subcritical_power_plant/subcritical_boiler.ipynb

@MAZamarripa, might also be able to help more.

**Reply by REI-Gardner** [reply]
@MAZamarripa is there a file that I could look at where the "Boiler Fire Side Model" has been implemented?

**Comment by MAZamarripa** [comment, marked as answer]
Hi, 
Here is the link to the documentation https://idaes-pse.readthedocs.io/en/stable/technical_specs/model_libraries/power_generation/unit_models/boiler_fireside.html

This model can be used in two ways, 1) user can provide fixed inputs for heat loss to the water wall; 2) user can provide a "surrogate model / correlation" to represent the heat loss to the water wall. 

Option 1: 
Example of the data dictionary: 
https://github.com/IDAES/idaes-pse/blob/main/idaes/power_generation/unit_models/tests/datadictionary.py

Example of the code being used for testing:
https://github.com/IDAES/idaes-pse/blob/main/idaes/power_generation/unit_models/tests/test_boilerfireside.py

Option 2: 
Example of data dictionary with surrogate models (developed off line):
https://github.com/IDAES/idaes-pse/blob/main/idaes/power_generation/flowsheets/subcritical_power_plant/generic_surrogate_dict.py

Example of a sub-critical coal fired boiler using the data dictionary (with surrogates):
https://github.com/IDAES/idaes-pse/blob/main/idaes/power_generation/flowsheets/subcritical_power_plant/subcritical_boiler_flowsheet.py

Example of a sub-critical coal fired power plant:
https://github.com/IDAES/idaes-pse/blob/main/idaes/power_generation/flowsheets/subcritical_power_plant/subcritical_power_plant.py

Surrogate models have been developed using a boiler fire-side rigorous simulation (c++ code). 
- A simple way to obtain an accurate estimate for heat loss, is to provide the steam temperature and flue gas outlet temperature, and solve problem to estimate the heat loss to the water wall (Qvar). 
- Another approach is to use the generic surrogate models provided in the example (generic boiler relative to a ~240 MWe plant). 
- Finally, we can provide access to the boiler fire-side rigorous simulation tool, however, the detailed information of the boiler (dimensions), burners (#, and levels, type), coal (type, specs, etc.) is required.
